analyzers/src/SonarAnalyzer.Common/Rules/AvoidDateTimeNowForBenchmarkingBase.cs

```diff
@@ -0,0 +1,68 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class AvoidDateTimeNowForBenchmarkingBase<TMemberAccess, TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TMemberAccess : SyntaxNode
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S6561";
+    protected override string MessageFormat => "Avoid using \"DateTime.Now\" for benchmarking or timing operations";
```

Comment:
The timing operations part is really bothering me, as it can be interpreted any way. I mean if you call AddHours isn't that a timing related operation? We should specify a bit more precisely what is not good there.
I unfortunately don't have a better idea at the moment. I will think about it.

Reply:
What about "Avoid using \"DateTime.Now\" for benchmarking or timespan calculation operations. ?